### PR TITLE
v5.2.18

### PR DESCRIPTION
**Pull request by CleanBread**

### CODE REVIEW

File: src/contexts/socket.tsx [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import React, { createContext, useCallback, useContext, useEffect, useRef, useState } from 'react'
 import { chainsData, type ConditionStatus } from '@azuro-org/toolkit'
 
-import { createBatch } from '../helpers'
+import { debounce } from 'src/helpers/debounce'
```
Comment (on47sky) [inline src/contexts/socket.tsx]: `src/` -> `../`

File: src/contexts/socket.tsx [hunk 2]
```diff
@@ -104,12 +105,85 @@ export const SocketProvider: React.FC<any> = ({ children }) => {
...
+    const run = (action: 'subscribe' | 'unsubscribe', conditionIds: string[]) => {
+      // group batch requests
+      const request = debounce(() => {
+        const subscribeQueue = [ ...actions.subscribe ]
+        const unsubscribeQueue = [ ...actions.unsubscribe ]
+
+        actions.subscribe = []
+        actions.unsubscribe = []
+
+        const weighs: Record<string, number> = {}
```
Comment (on47sky) [inline src/contexts/socket.tsx]: typo, `weights` :)

**Review by on47sky — APPROVED** [review]

File: src/contexts/socket.tsx (hunk 2 above)
Comment (CleanBread) [reply src/contexts/socket.tsx]: 3d82f83efba55776d733c83f501bf4f20daadaab

File: src/contexts/socket.tsx (hunk 1 above)
Comment (CleanBread) [reply src/contexts/socket.tsx]: 3d82f83efba55776d733c83f501bf4f20daadaab

**Pull request merged**